Broken CI target for freebsd nightly

Since https://github.com/rust-lang/rust/pull/120869 broken CI target for freebsd nightly with follows:
```
Run cargo hack test --locked --release --feature-powerset --exclude-features wasm
  cargo hack test --locked --release --feature-powerset --exclude-features wasm
  shell: /bin/bash -e {0}
  env:
    RUST_BACKTRACE: 1
/home/vagrant/actions-runner-pna/_work/_temp/1ec92e66-c6ba-41a1-bb7f-6fccd773490f.sh: line 1: 12153 Segmentation fault      (core dumped) cargo hack test --locked --release --feature-powerset --exclude-features wasm
Error: Process completed with exit code 139.
```

https://github.com/ChanTsune/Portable-Network-Archive/actions/runs/11460996690/job/31888982687

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -17,17 +17,6 @@ jobs:
           - stable
           - beta
           - nightly
-          - nightly-2024-10-21 # FIXME: unpin this
-        exclude:
-          - os: ubuntu-latest
-            rust: nightly-2024-10-21
-          - os: windows-latest
-            rust: nightly-2024-10-21
-          - os: macos-latest
-            rust: nightly-2024-10-21
-          - os: freebsd-latest
-            rust: nightly
-
     runs-on: ${{ matrix.os }}
     steps:
       - name: git autocrlf false

